src/grid/atomgrid.py

```diff
@@ -288,6 +291,113 @@ def convert_cart_to_sph(self, points=None, center=None):
         center = self.center if center is None else np.asarray(center)
         return convert_cart_to_sph(points, center)
 
+    def fit_values(self, value_array):
+        """Fit given value arrays into splines that matches atomic grid.
+
+        Parameters
+        ----------
+        value_array : np.ndarray(N,)
+            a 1d-array evaluated at each atomic grid point
+
+        Returns
+        -------
+        list[scipy.PPoly]
+            A list of cubic spline fitted by given value arrays
+        """
+        if self._basis is None:
```

Comment:
Can you add the check at the beginning of this method to make sure `value_array` has `(self.size,)` shape?